torch/csrc/jit/passes/python_print.cpp

```diff
@@ -84,48 +88,33 @@ static bool isValidIdentifier(const std::string& name) {
   return true;
 }
 
-// handles names of the form, e.g., self.a.b
-// if a field is not a valid identifier, then it will print as, e.g.
-// getattr(self, "0").b
-struct QualifiedName;
-using QualifiedNamePtr = c10::intrusive_ptr<QualifiedName>;
-struct QualifiedName : c10::intrusive_ptr_target {
-  QualifiedName(QualifiedNamePtr prefix, std::string name)
-      : prefix_(std::move(prefix)), name_(std::move(name)) {}
-  QualifiedNamePtr prefix_;
-  std::string name_;
-  static QualifiedNamePtr create(QualifiedNamePtr prefix, std::string name) {
-    return c10::make_intrusive<QualifiedName>(
-        std::move(prefix), std::move(name));
-  }
-  static QualifiedNamePtr create(std::string name) {
-    return c10::make_intrusive<QualifiedName>(
-        QualifiedNamePtr(), std::move(name));
-  }
-  std::string str() const {
-    std::stringstream ss;
-    emit(ss);
-    return ss.str();
-  }
-
- private:
-  void emit(std::ostream& out) const {
-    if (isValidIdentifier(name_)) {
-      if (prefix_) {
-        prefix_->emit(out);
-        out << ".";
-      }
-      out << name_;
-    } else {
-      AT_ASSERT(prefix_);
-      out << "getattr(";
-      prefix_->emit(out);
-      out << ", ";
-      printQuotedString(out, name_);
-      out << ")";
+static void emitQualifiedName(std::ostream& out, QualifiedNamePtr name) {
+  const auto& name_ = name->name_;
+  const auto& prefix_ = name->prefix_;
+  if (isValidIdentifier(name_)) {
+    if (prefix_) {
+      emitQualifiedName(out, prefix_);
+      out << ".";
     }
+    out << name_;
+  } else {
+    AT_ASSERT(prefix_);
```

Comment:
I'm not sure i understand what is happening here

Reply:
Just preserving what's already here. I assume this is because invalid idents can only come from Sequential, which is guaranteed to have a prefix